### PR TITLE
Added dns_config with info and integration tests

### PR DESCRIPTION
**Pull request by anazobec**
Created a module for DNSConfig: ``dns_config.py`` (in ``module_utils/`` and ``modules/``), ``dns_config_info.py`` (in ``modules/``) and integration tests for the newly created module ``dns_config_info.py`` and ``dns_config.py``

### CODE REVIEW

**Comment by justinc1** [comment]
PR assignee - you and reviewer - me should be set.
CI tests should pass.

File: tests/integration/targets/dns_config_info/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+- environment:
+    SC_HOST: "{{ sc_host }}"
+    SC_USERNAME: "{{ sc_username }}"
+    SC_PASSWORD: "{{ sc_password }}"
+
+  block:
+    - name: Retrieve info about DNS Config
```
Comment (justinc1) [inline tests/integration/targets/dns_config_info/tasks/main.yml:8]: We can use api module to push server into known state, then we can check if exactly correct values are returned. 

What I hate is we need to restore server to original state - but is a test system for CI tests, so this restore is a minor thing.

File: tests/integration/targets/dns_config/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          - "test2"
+          - "test3"
+          - "test4"
+        dns_servers:
+          - "1.2.3.4"
+        state: present
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result.diff.before != result.diff.after
```
Comment (justinc1) [inline tests/integration/targets/dns_config/tasks/main.yml:22]: Check also result.records is correct, and same for result.changed. Those two are usually used later in playbook.

The result.diff is usually used only for debugging - when you look into verbose output on your screen, so most other integration tests do not check in depth what is inside (the != check is a bonus, so please keep it).

File: tests/integration/targets/dns_config/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          - "test4"
+        dns_servers:
+          - "1.2.3.4"
+        state: absent
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result.diff.before != result.diff.after
+          - result.diff.after.search_domains == []
+          - result.diff.after.server_ips == ["1.1.1.1", "1.0.0.1"]
```
Comment (justinc1) [inline tests/integration/targets/dns_config/tasks/main.yml:43]: hardcoded values suitable only for one specific test cluster.

**Review by justinc1 — CHANGES_REQUESTED** [review]
We can have a call.

File: examples/dns_config.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+---
+- name: Example dns_config module
+  hosts: localhost
+  connection: local
+  gather_facts: false
+
+  tasks:
+    # ------------------------------------------------------
+    - name: Add to DNS configuration records
```
[the remark this answers is not in the document]
Comment (anazobec) [reply examples/dns_config.yml:9]: yes, I'm up for a call

File: plugins/modules/dns_config.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+    TaskTag.wait_task(rest_client, task_tag_create)
+    new_dns_config = DNSConfig.get_by_uuid(module.params, rest_client)
+    after = new_dns_config.to_ansible()
+    changed, result, _dict = before != after, [after], dict(before=before, after=after)
+
+    return changed, result, _dict
+
+
+def ensure_absent(module, rest_client):  # delete
```
[the remark this answers is not in the document]
Comment (anazobec) [reply plugins/modules/dns_config.py:71]: ``ensure_absent`` is not removing the whole DNS config. It's only removing items from a list inside the DNS config. That list can either be searchDomains or dnsServers. So the DNS config itself is never removed. Unless I'm not supposed to do that?

**Comment by justinc1** [comment]
One more thing - we do need to include also fragment for future release notes. Something like https://github.com/ScaleComputing/HyperCoreAnsibleCollection/blob/main/changelogs/fragments/release_110.yml

File: tests/integration/targets/dns_config/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+          - ips: []
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result.diff.before != result.diff.after
+          - result.new_state.0.search_domains == []
+          - result.new_state.0.server_ips == []
+          - result.new_state.0.uuid == actual_uuid
+
+# -----------------------------------------------------------
```
Comment (justinc1) [inline tests/integration/targets/dns_config/tasks/main.yml:153]: Another corner case:
```
       overwrite: all
        search_domains:
          - names:
              - example.domain1.com
              - example.domain2.com
            prepend: True
```

Here module should fail, because combo overwrite/prepend is invalid.

Do you want (need) to have different prepend for search_domains and for dns_servers? No, it is not needed - we can invoke module twice, with different prepend value.


What about a single parameter, name like `state` (`mode` as second alternative name), possible values `set` (means overwrite), `before` (means prepend), `after` (means add after)?

Since we are able to have "partial add", we will in future likely want to have also "partial remove" (no need to implement this now; but with `state=absent`

We do have something similar in module vm_disk.py - `state` can be `present` or `absent` (those two values are very common for state), in addition we have unusual `set` value - description is "With I(state=set), the VM is reconfigured to have exactly such disks as specified".


Another corner case to test: old DNS IPs are `a, b, c`. You can now try to say `prepend`:
- `d, e`
- `a, b`
- `b, a` - is desired new state `a,b,c` or `b,a,c`? Likely it is not relevant which one is selected, but it should be tested, so that behaviour does not magically change when something will be changed in code.
- `b, c`
- `c, b`
I know there is some set() to remove duplicates, but it is not tested (no unit tests).

**Review by justinc1 — CHANGES_REQUESTED** [review]

File: examples/dns_config.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        state: present
+
+    # ------------------------------------------------------
+    - name: Delete DNS configuration records
+      scale_computing.hypercore.dns_config:
+        search_domains:
+          - do_not_need_this_dns
+        dns_servers:
+          - 5.6.7.8
+        state: absent
```
[the remark this answers is not in the document]
Comment (anazobec) [reply examples/dns_config.yml]: pozabila sem example direktorij dodat v commit

File: examples/dns_config_info.yml [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+  tasks:
+  # ------------------------------------------------------
+    - name: List DNS configuration
+      scale_computing.hypercore.dns_config_info:
+      register: dns_config_info_results
+
+    - name: Show all configs
+      debug:
+        var: is_results
```
[the remark this answers is not in the document]
Comment (anazobec) [reply examples/dns_config_info.yml]: I forgot to add the examples directory to the commit. Just fixed that :)

File: plugins/modules/dns_config.py [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+short_description: Modify DNS configuration on HyperCore API
+description:
+  - Use this module to add to or delete from a DNS configuration on HyperCore API.
+version_added: 1.1.1
+extends_documentation_fragment:
+  - scale_computing.hypercore.cluster_instance
+seealso: 
+  - module: scale_computing.hypercore.dns_config_info
+options:
+  overwrite:
```
[the remark this answers is not in the document]
Comment (anazobec) [reply plugins/modules/dns_config.py:20]: I forgot to add the examples directory to the commit. Just fixed that :)

[the remark this answers is not in the document]
Comment (anazobec) [reply plugins/modules/dns_config.py:20]: I forgot to add the examples directory to the commit. Just fixed that :)

**Review by justinc1 — CHANGES_REQUESTED** [review]
Only minor changes.

File: plugins/modules/dns_config_info.py [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+EXAMPLES = r"""
+- name: List all configurations on DNS configuration on HyperCore API
+  scale_computing.hypercore.dns_config_info:
+  register: dns_config
+"""
+
+RETURN = r"""
+records:
+  description:
+    - A list of DNS configuration records.
```
Comment (justinc1) [inline plugins/modules/dns_config_info.py:36]: Module actually returns a single record, not a list.

Change to 
```
record:
  description:
    - DNS configuration record.
```

**Review by justinc1 — CHANGES_REQUESTED** [review]
One minor change

**Review by justinc1 — APPROVED** [review]
Thank you ! :)
